File: problems/26-remove-duplicates-from-sorted-array.ts
export function removeDuplicates(nums: Array<number>): number {
  if (!nums.length || nums.length === 1) return nums.length;

  // nums.splice(0, nums.length, ...new Set(nums));

  const map: Record<number, string> = {};
  for (let i = nums.length - 1; i >= 0; i--) {
    if (nums[i] in map) {
      nums.splice(i, 1);
      continue;
    }
    map[nums[i]] = "";
  }

  return nums.length;
}

File: go/problems/26_remove_duplicates_from_sorted_array.go
package main

func removeDuplicates(nums []int) int {
	if len(nums) < 1 {
		return 0
	}

	hashMap := make(map[int]struct{})
	writeIndex := 0
	for _, num := range nums {
		if _, ok := hashMap[num]; !ok {
			hashMap[num] = struct{}{}
			nums[writeIndex] = num
			writeIndex++
		}
	}
	return len(hashMap)
}
